File: services/clinical-service/src/domain/events/EncounterCreatedEvent.ts
export interface DomainEvent {
    eventType: string;
    occurredAt: Date;
    payload: Record<string, unknown>;
}

export class EncounterCreatedEvent implements DomainEvent {
    readonly eventType = 'encounter.created';
    readonly occurredAt: Date;
    readonly payload: Record<string, unknown>;

    constructor(encounterId: string, patientId: string) {
        this.occurredAt = new Date();
        this.payload = { encounterId, patientId };
    }
}

File: services/clinical-service/src/main/java/com/medtrust/clinical/domain/event/EncounterCreatedEvent.java
package com.medtrust.clinical.domain.event;

import java.time.Instant;
import java.util.Map;

public record EncounterCreatedEvent(
        String encounterId,
        String patientId) implements DomainEvent {

    @Override
    public String eventType() {
        return "encounter.created";
    }

    @Override
    public Instant occurredAt() {
        return Instant.now();
    }

    @Override
    public Map<String, Object> payload() {
        return Map.of(
                "encounterId", encounterId,
                "patientId", patientId);
    }
}
